library/public/camera.h

```diff
@@ -24,25 +25,39 @@ namespace f3d
 class F3D_EXPORT camera
 {
 public:
-  typedef std::array<double, 3> vector3_t;
-  typedef std::array<double, 16> matrix4_t;
+  //@{
+  /**
+   * Camera specific types.
+   */
+  struct point3_t : std::array<double, 3>
```

Comment:
This is more along the lines of strong typing, which will enable the compiler to prevent doing things like accidentally using a vector when meaning to use a point. There is more type safety.

Reply:
Yup, that would be useless without the struct/class

Reply:
Ok, then it should probably goes in its own file, or at least move it outside of the camera class scope, so we can reuse these types in other places later.

Reply:
good point

Reply:
Agree with its own file outside of camera. I would think you would also want to add some math operators and things like `operator<<`.

Reply:
it may be added later, this feels a bit outside of the scope of the first version imo.